packages/strapi-plugin-i18n/config/functions/migrations/field.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const { difference, pick, orderBy, prop, intersection } = require('lodash/fp');
+const { getService } = require('../../../utils');
+
+const BATCH_SIZE = 1000;
+
+const shouldBeProceed = processedLocaleCodes => entry =>
+  entry.localizations.length > 1 &&
+  intersection(entry.localizations.map(prop('locale')), processedLocaleCodes).length === 0;
+
+const migrateForBookshelf = async ({ ORM, model, attributesToMigrate, locales }) => {
+  // Create tmp table with all updates to make (faster than make updates one by one)
+  const TMP_TABLE_NAME = '__tmp__i18n_field_migration';
+  const columnsToCopy = ['id', ...attributesToMigrate];
+  await ORM.knex.schema.dropTableIfExists(TMP_TABLE_NAME);
+  await ORM.knex.raw(`CREATE TABLE ?? AS ??`, [
+    TMP_TABLE_NAME,
+    ORM.knex
+      .select(columnsToCopy)
+      .from(model.collectionName)
+      .whereRaw('?', 0),
+  ]);
+
+  // Transaction is started after DDL because of MySQL (https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html)
```

Comment:
Because the transaction doesn't wrap all the function which would have been better. So I explain why. Do you think it would be better without this comment?